### DISCUSSION TITLE
TFT_eSPI compilation issue using vscode with Windows

### DISCUSSION THREAD

**Discussion by Shikamru** [Bodmer/TFT_eSPI#3139] [Raspberry Pi Pico]
Hello Bodmer,
                      I am using pico w board with c/c++ sdk for my projects, I wanted to import the arduino libraries into my workspace so i found this repo pico-arduino-compat [https://github.com/kholia/pico-arduino-compat](url) ,  which helped me to do that. however which including the tft liraries am facing a lot of errors like below, am i missing anything here, kindly guide me .

`error: 'TFT_eSPI_Button' does not name a type
[build]     5 | TFT_eSPI_Button::TFT_eSPI_Button(void) {
[build]       | ^~~~~~~~~~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:14:6: error: 'TFT_eSPI_Button' has not been declared
[build]    14 | void TFT_eSPI_Button::initButton(
[build]       |      ^~~~~~~~~~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:14:6: error: variable or field 'initButton' declared void
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:2: error: 'TFT_eSPI' was not declared in this scope
[build]    15 |  TFT_eSPI *gfx, int16_t x, int16_t y, uint16_t w, uint16_t h,
[build]       |  ^~~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:12: error: 'gfx' was not declared in this scope
[build]    15 |  TFT_eSPI *gfx, int16_t x, int16_t y, uint16_t w, uint16_t h,
[build]       |            ^~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:17: error: 'int16_t' was not declared in this scope
[build]    15 |  TFT_eSPI *gfx, int16_t x, int16_t y, uint16_t w, uint16_t h,
[build]       |                 ^~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:1:1: note: 'int16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
[build]   +++ |+#include <cstdint>
[build]     1 | /***************************************************************************************
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:28: error: 'int16_t' was not declared in this scope
[build]    15 |  TFT_eSPI *gfx, int16_t x, int16_t y, uint16_t w, uint16_t h,
[build]       |                            ^~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:28: note: 'int16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:39: error: 'uint16_t' was not declared in this scope
[build]    15 |  TFT_eSPI *gfx, int16_t x, int16_t y, uint16_t w, uint16_t h,
[build]       |                                       ^~~~~~~~
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:39: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
[build] C:/picow_setup/pico-arduino-compat/libs/tft_espi/TFT_eSPI/Extensions/Button.cpp:15:51: error: 'uint16_t' was not declared `
these are few.

note ->  I am using vs code on windows. 

regards,
shikamaru

**Comment by Bodmer** [comment]
The library is intended for the Arduino IDE environment so I am not able to help with other build setups.

**Reply by rozrabiak** [reply]
I'm using vscode and platformio and compiles without any errors.
Try to update libraries, platforms and try again.
